script/helpers/Cosmos-Emulator.ps1

```diff
@@ -0,0 +1,146 @@
+Exact path to Microsoft.Azure.Cosmos.Emulator.exe
+
+.PARAMETER Stage
+Determines what part of the script to run. Has to be either Install or Launch
+#>
+[CmdletBinding()]
+Param (
+    [string] $EmulatorMsiUrl = "https://aka.ms/cosmosdb-emulator",
+    [string] $StartParameters,
+    [string] $Emulator,
+    [Parameter(Mandatory = $True)]
+    [ValidateSet('Install', 'Launch')]
+    [string] $Stage
+)
+
+$targetDir = Join-Path $Env:Temp AzureCosmosEmulator
+$logFile = Join-Path $Env:Temp log.txt
+$productName = "Azure Cosmos DB Emulator"
+
+if ([string]::IsNullOrEmpty($Emulator)) {
+    $Emulator = (Join-Path $targetDir (Join-Path $productName "Microsoft.Azure.Cosmos.Emulator.exe"))
+}
+
+if ($Stage -eq "Install") {
+    $downloadTryCount = 0
+    New-Item $targetDir -Type Directory
+    New-Item $logFile -Type File
+    do {
+        # Download and Extract Public Cosmos DB Emulator
+        Write-Host "Downloading and extracting Cosmos DB Emulator - $EmulatorMsiUrl"
+        Write-Host "Target Directory $targetDir"
+        Write-Host "Log File $logFile"
+
+        $downloadTryCount++
+        Write-Host "Download Try Count: $downloadTryCount"
+        Remove-Item -Path (Join-Path $targetDir '*') -Recurse
+        Clear-Content -Path $logFile
+
+        $installProcess = Start-Process msiexec -Wait -PassThru -ArgumentList "/a $EmulatorMsiUrl TARGETDIR=$targetDir /qn /liew $logFile"
+        Get-Content $logFile
+        Write-Host "Exit Code: $($installProcess.ExitCode)"
+    }
+    while (($installProcess.ExitCode -ne 0) -and ($downloadTryCount -lt 3))
+
+    if (Test-Path (Join-Path $Env:LOCALAPPDATA CosmosDbEmulator)) {
+        Write-Host "Deleting Cosmos DB Emulator data"
+        Remove-Item -Recurse -Force $Env:LOCALAPPDATA\CosmosDbEmulator
+    }
+
+    Write-Host "Getting Cosmos DB Emulator Version"
+    $fileVersion = Get-ChildItem $Emulator
+    Write-Host $Emulator $fileVersion.VersionInfo
+}
+
+if ($Stage -eq "Launch") {
+    Write-Host "Launching Cosmos DB Emulator"
+    if (!(Test-Path $Emulator)) {
+        Write-Error "The emulator is not installed where expected at '$Emulator'"
+        return
+    }
+
+    $process = Start-Process $Emulator -ArgumentList "/getstatus" -PassThru -Wait
+    switch ($process.ExitCode) {
+        1 {
+            Write-Host "The emulator is already starting"
+            return
+        }
+        2 {
+            Write-Host "The emulator is already running"
+            return
+        }
+        3 {
+            Write-Host "The emulator is stopped"
+        }
+        default {
+            Write-Host "Unrecognized exit code $($process.ExitCode)"
+            return
+        }
+    }
+
+    $argumentList = ""
+    if (-not [string]::IsNullOrEmpty($StartParameters)) {
+        $argumentList += , $StartParameters
+    }
+    else {
+        # Use the default params if none provided
+        $argumentList = "/noexplorer /noui /enablepreview /disableratelimiting /enableaadauthentication"
```

Comment:
FYI, this is the argumentList that we use in other SDKs: 

`/noexplorer /noui /enablepreview /EnableSqlComputeEndpoint /SqlComputePort=9999 /disableratelimiting /partitioncount=50 /consistency=Strong`

reference: https://github.com/Azure/azure-sdk-for-java/blob/main/eng/pipelines/templates/stages/cosmos-sdk-client.yml#L101

Reply:
Go uses different args (https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/data/azcosmos/ci.yml#L46) so I'm not sure there's a ton of consistency there 😅 . These args are working for now and we can override them easily in the YAML :)